### PR TITLE
feat: add `size` prop to Sidebar

### PR DESCRIPTION
**Pull request by TomasSlama**
[FX-3279]

### Description

- [Figma](https://www.figma.com/file/5SCTOPrCDcHuk5We091GBn/Product-Library?node-id=408%3A15410&t=WvXC7OMI8HNnEpAF-0)

When working on the task I noticed, that we don't follow the BASE width of the Sidebar, so I have created a new prop `size`. By default `medium` has the same size as it had before, so there is no breaking change.

### How to test

- check t[he temploy](https://picasso.toptal.net/sidebar-size/?path=/story/components-pagesidebar--pagesidebar#sizes)

### Screenshots

image

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Make sure that additions and changes on design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [x] Annotate all `props` in component with documentation
- [x] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [x] Covered with tests

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-3279]: https://toptal-core.atlassian.net/browse/FX-3279?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by ozgurkececioglu — APPROVED** [review]
Looks good 👍

**Review by dmaklygin — APPROVED** [review]
Checked the code. Looks good!

**Review by LashaJini — APPROVED** [review]
Checked the demo, looks good :+1:

File: packages/picasso/src/PageSidebar/story/Size.example.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      <Page.Sidebar.Item icon={<Participants16 />}>Users</Page.Sidebar.Item>
+    </Page.Sidebar.Menu>
+  </Page.Sidebar>
+)
+
+const Example = () => (
+  <Container flex gap='large'>
+    <Container>
+      <Typography variant='heading' size='xsmall'>
+        Small
```
[the remark this answers is not in the document]
Comment (TomasSlama) [reply packages/picasso/src/PageSidebar/story/Size.example.tsx]: ✅

**Pull request merged**